cpp/arcticdb/version/version_core.cpp

```diff
@@ -847,11 +853,15 @@ void check_incompletes_index_ranges_dont_overlap(const std::shared_ptr<PipelineC
         // Use ordered set so we only need to compare adjacent elements
         std::set<TimestampRange> unique_timestamp_ranges;
         for (auto it = pipeline_context->incompletes_begin(); it!= pipeline_context->end(); it++) {
+            if (it->slice_and_key().slice().rows().diff() == 0) {
+                continue;
+            }
             sorting::check<ErrorCode::E_UNSORTED_DATA>(
                     !last_existing_index_value.has_value() || it->slice_and_key().key().start_time() >= *last_existing_index_value,
                     "Cannot append staged segments to existing data as incomplete segment contains index value < existing data (in UTC): {} <= {}",
                     date_and_time(it->slice_and_key().key().start_time()),
-                    date_and_time(*last_existing_index_value));
+                    last_existing_index_value ? date_and_time(*last_existing_index_value) : "NaT"
```

Comment:
This doesn't make sense. This exception will only be throw if `last_existing_index_value.has_value()` is true

Reply:
That is true, but we're passing function arguments and by standard they are always evaluated and the order of evaluation is not defined. So even if the condition for throwing is false the rest of the arguments will be evaluated including the call to `*last_existing_index_value` which throws when the optional is empty.

Reply:
Ah I see. In that case I would replace the `NaT` with an empty string and add a comment about it being unreachable but needed for this reason